### PR TITLE
Reworked AWS launch_configuration to launch_templates

### PR DESCRIPTION
**Pull request by filipvh-sentia**
I tried to create identical launch template to the launch configuration you previously had. Launch templates are preferred over launch configuration per the AWS documentation ( https://docs.aws.amazon.com/autoscaling/ec2/userguide/launch-configurations.html ).

I didn't specify a disk size so it should take the default provided by the AMI ( which was 8Gb if I recall correctly ).
Default disk type is now explicitly GP3 with the minimum values.

I set `http_endpoint = "enabled"` in every `metadata_options`. This prevents a new version being created every time you execute the terraform code.

Let me know if you have any remarks on the code. I kept all the parameters in the same order, but feel  `key_name = var.key_name` and `ebs_optimized = true` would be easier to spot if I add them with the other parameters ( between `instance_type` and `user_data` ).

### CODE REVIEW

**Comment by hugoShaka** [comment]
@filipvh-sentia thank you for the PR and the heads up. We'll definitely want to fix that. The PR looks good a priori but I'll need to run a couple of tests before approving. I'll try to do it by the end of next week.

I'm not sure whether we want to backport this change to v12. Some recent projects won't be able to run v12.1 because of this, but it feels like a potentially large breaking change that should be part of Teleport 13.

**Review by hugoShaka — APPROVED** [review]
I tested the PR, and the change is seamless. New instances are launched with the new template; as expected, existing instances are unaffected.

**Review by r0mant — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by hugoShaka** [comment]
Hi @filipvh-sentia ,
Thanks for your contribution. We are currently unable to run our tests on
external contributions. As such, I'll be helping you validate and
merge your pull request. I've created an internal PR here:

https://github.com/gravitational/teleport/pull/24675

Please wait for another approver and I'll get your code merged.

**Comment by hugoShaka** [comment]
Merged in: https://github.com/gravitational/teleport/commit/00f99f478803253a467367429af3366c7ebd9681

**Comment by filipvh-sentia** [comment]
Hi Hugo!

Glad to have helped! :)

**Pull request closed without merging**
